examples/archive.html

```diff
@@ -106,7 +106,27 @@ <h3 id="offcanvasRightLabel">Images</h3>
                 let placeButton = document.createElement('a');
 
                 placeButton.classList.add('btn', 'btn-sm', 'btn-outline-secondary', 'place-button');
-                placeButton.innerHTML = 'Place on map';
+                placeButton.innerHTML = 'Place on map ';
+
+                placeButton.addEventListener('click', (event) => {
+                  if(!placeButton.getElementsByClassName('fa-spinner')[0]){
+                    const spinner = document.createElement('i')
+                    spinner.setAttribute('class', "fa fa-spinner fa-spin")
+                    spinner.setAttribute('font-family', 'font awesome')
+  
+                    if (event.target.classList.contains('place-button')){
```

Comment:
@RealRichi3 @jywarren, since the click event has been placed directly on the placeButton,  could we get rid of the check on line 117?

Reply:
you mean we should add the spinner without the check?

Reply:
Yes. As opposed to the earlier version  `document.addEventListener('click', (event)=>{...` , there is no event delegation in this. The event has been placed directly to the target.

Reply:
@RealRichi3 We have a repetition of code. check lines 157 to 161, that is where I had earlier suggested the spinner go.

Reply:
Deleting lines 157-161 should clean it up right?

Reply:
@RealRichi3 it would. However, performance-wise, it won't be effective since we are creating an event on every button. I still prefer the one I suggested. This is what my solution looks like:


       document.addEventListener("click", (event) => {
        if (event.target.classList.contains("place-button")) {
        const targetPlaceButton = event.target;
        const spinner = document.createElement('i');
        spinner.classList.add('fa', 'fa-spinner', 'fa-spin');
        targetPlaceButton.append(spinner);

        let imageURL = event.target.previousElementSibling.src;
        let image = L.distortableImageOverlay(imageURL);
        map.imgGroup.addLayer(image);
        image.addEventListener("load", () => {
          spinner.remove()
        });
      }
    });

Reply:
Thanks @sainamercy, i added an extra check incase a user clicks the button multiple times, this will add more than one spinner icon. But it's all good now

Reply:
I think you are getting the error coz it's wrapped around the `forEach` block. Try outside the block, maybe where it was before, then see if there's a difference.

Reply:
Ohh, i understand now, but i think it looks good now. @jywarren please review